cost_rules/app.py

```diff
@@ -252,22 +313,20 @@ def lambda_handler(event, context):
 
     result = { "statusCode": 200 }
     try:
-        # get environment variables
-        chart_url = _get_os_var('ChartOfAccountsURL')
-
-        _tag_list = _get_os_var('ProgramCodeTagList')
-        tag_list = _parse_env_list(_tag_list)
+        if 'path' in event:
+            event_path = event['path']
 
-        # get account tags
-        account_codes = collect_account_tag_codes(tag_list)
+            if event_path == '/program-codes':
+                rules_data = list_program_rules()
 
-        # get chart of accounts
-        chart_data = collect_chart_of_accounts(chart_url)
+            elif event_path == '/owner-emails':
+                rules_data = list_owner_rules()
```

Comment:
why not create a function per path? that might decouple one function from another and simplify your code.

you can do this in `template.yaml`

for program codes:
```
Handler: cost_rules/program_codes.lambda_handler
```

for owner email:
```
Handler: cost_rules/owner_emails.lambda_handler
```

Reply:
I'm not sure what you mean by decoupling the functions or how to further simplify the code, but I guess I can create a couple SAM Functions with dedicated submodules for the two endpoints, though that's mostly just moving the functions into different files, they'll still all depend on the `_build_*()` functions in a shared submodule for the heavy lifting since the processing is so similar